docs/dg/dev/troubleshooting/troubleshooting-general-technical-issues/session-locking-issues.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Session Locking Issues
+description: Learn how to troubleshoot and resolve session locking issues in Spryker-based projects.
+last_updated: Jun 27, 2025
+template: troubleshooting-guide-template
+originalArticleId: c234b12e-5260-4f86-97b4-44c7ef5c8dbf
+---
+
+# Optimizing Session Handling for High-Traffic Pages
+
+This guide outlines a common performance issue related to session handling on high-traffic websites and provides both immediate mitigation strategies and a long-term architectural solution.
+
+## The Problem: Session Locking on Read-Only GET Requests
+
+A frequent performance bottleneck, especially under heavy crawler or bot traffic, is session contention in Redis.
+
+### Core Issue
+
+By default, Spryker may initiate a session and apply a lock for every page request, including simple GET requests. This is primarily due to the on-page generation of CSRF tokens for forms (e.g., "Add to Cart" forms on a product page, newsletter sign-ups). The generation of this token is a session "write" operation.
```

Comment:
as I remember, default session handler with lock is only on Yves or Backoffice. Mb it worth to mention it here?

Reply:
I get that this is clear from content that this is Yves related, but still